platforms/jvm/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/CrossTaskIncrementalJavaCompilationIntegrationTest.groovy

```diff
@@ -162,7 +164,7 @@ abstract class CrossTaskIncrementalJavaCompilationIntegrationTest extends Abstra
             tasks.compileJava {
                 modularity.inferModulePath = false
                 options.compilerArgs << "--module-path=\${classpath.join(File.pathSeparator)}" \
-                    << "--module-source-path" << file("src/main/$languageName")
+                    << "--module-source-path" << file("src/main/$languageName").absolutePath
```

Comment:
❓ Is this change necessary now?

Reply:
How does `<<` work here, but had to be replaced in other places?

Reply:
Actually I replaced it everywhere at first, but after support for `<<` was merged I reverted it in integration tests. But I kept the change in samples/snippets, since to me as Java developer it's more obvious to use `add` than `<<`. 

But I can also revert it in docs and we can fix it on master if we find a need to

Reply:
>❓ Is this change necessary now?

Yes it is, since we don't have automatic type translation from File -> String (yet) for providers

Reply:
I reverted changes of << in snippets/samples